test-utils/test-framework/src/framework.rs

```diff
@@ -0,0 +1,116 @@
+use actix_rt;
+use anyhow;
+use std::future::Future;
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, Mutex};
+
+use crate::YagnaMock;
+
+pub use macros::prepare_test_dir;
+pub use ya_framework_macro::framework_test;
+
+#[derive(Clone)]
+pub struct YagnaFramework {
+    inner: Arc<Mutex<YagnaNetworkImpl>>,
+    test_dir: PathBuf,
+    test_name: String,
+}
+
+/// Entities that require tear down after test is finished.
+/// This allows us to kill yagna process even in case of panic.
+#[derive(Clone)]
+pub struct YagnaNetworkImpl {
+    nodes: Vec<YagnaMock>,
+}
+
+pub mod macros {
+    #[macro_export]
+    macro_rules! prepare_test_dir {
+        () => {
+            // CARGO_TARGET_TMPDIR is available in compile time only in binary modules, so we can't
+            // use it in this library. Thanks to using macro it will be resolved in final code not here
+            // and it will work.
+            ya_test_framework::framework::prepare_test_dir_(env!("CARGO_TARGET_TMPDIR"))
```

Comment:
Tests are compiled into binaries, so it should work.
I don't want `temp_dir`, exactly for this reason that it creates directory in `/tmp`. If you are using any developer environment, `CARGO_TARGET_TMPDIR` will be inside your project and `/tmp` requires additional steps to find it.
Goth is good example how cumbersome it can be